visualization/app/codeCharta/util/treeMapGenerator.ts

```diff
@@ -25,56 +38,80 @@ export function createTreemapNodes(map: CodeMapNode, state: State, metricData: N
-	const scale = (state.treeMap.mapSize * 2 + getEstimatedNodesPerSide(hierarchyNode) * state.dynamicSettings.margin) / nodes[0].length
+	for (const fixedFolder of hierarchyNode.children) {
 
-	scaleRoot(nodes[0], scale)
+		const squarified = getSquarifiedTreeMap(fixedFolder.data, state)
 
-	for (const fixedFolder of map.children) {
-		const squarified = getSquarifiedTreeMap(fixedFolder, state)
 		for (const squarifiedNode of squarified.treeMap.descendants()) {
-			scaleAndTranslateSquarifiedNode(squarifiedNode, fixedFolder, squarified, scale)
+			// squarified.width/height is a sum of the fixedPosition.width/height and applied margins
+			// The following scaling factors are used later to calculate out the original percentage fixed width/height.
+			// Example: Width and Height fixed at 20. Added margins 42px (absolute value) for each = 62px
+			//  20/62 = 0.322...       62 x 0.322... = 20 :)
+			const scaleX = fixedFolder.data.fixedPosition.width / squarified.width
+			const scaleY = fixedFolder.data.fixedPosition.height / squarified.height
+
+			// Scale the fixed (percentage) positions of a node by the right scale factor, so that it will be placed properly.
+			// The squarifiedNode coordinates x0, x1, y0, y1 are already assigned with positions from the treemap algorithm.
+			squarifiedNode.x0 = (squarifiedNode.x0 * scaleX + fixedFolder.data.fixedPosition.left) * scaleWidth
+			squarifiedNode.x1 = (squarifiedNode.x1 * scaleX + fixedFolder.data.fixedPosition.left) * scaleWidth
+			squarifiedNode.y0 = (squarifiedNode.y0 * scaleY + fixedFolder.data.fixedPosition.top) * scaleLength
+			squarifiedNode.y1 = (squarifiedNode.y1 * scaleY + fixedFolder.data.fixedPosition.top) * scaleLength
+
+			// Add x and y (absolute px) offsets from parent fixed folder, if any.
+			squarifiedNode.x0 += offsetX0
+			squarifiedNode.x1 += offsetX0
+			squarifiedNode.y0 += offsetY0
+			squarifiedNode.y1 += offsetY0
+
 			const node = TreeMapHelper.buildNodeFrom(squarifiedNode, heightScale, maxHeight, state, isDeltaState)
 			nodes.push(node)
+
+			if (hasFixedFolders(fixedFolder.data)) {
+				// Imagine the parent Folder has absolute px-width of 341px
+				// We need to calculate a scale factor for the fixed child to transform a relative percentage fixed width/height to px.
+				// Example: 20% of 341px:
+				//  fixedWidth=20, parent Fixed Folder width: 341px => 20 * (scaleLength=341/100)
+				// The original scaleWidth/scaleLength must be retained for the case that you have more than one fixedFolder on the same level.
+				const childRelativeLengthScale = node.length / 100
+				const childRelativeWidthScale = node.width / 100
+
+				Array.prototype.push.apply(
+					nodes,
+					buildSquarifiedTreeMapsForFixedFolders(fixedFolder, state, childRelativeLengthScale, childRelativeWidthScale, squarifiedNode.x0, squarifiedNode.y0, heightScale, maxHeight, isDeltaState)
```

Comment:
Minor: I think using the spread operator would make it look nicer here :)
https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Operators/Spread_syntax

Reply:
What do you think @DmytroHutkin ? :)